File: server/project/mj-client/src/main/java/majiang/client/portal/admin/form/PageForm.java
package majiang.client.portal.admin.form;

import org.forkjoin.apikit.core.Message;
import org.hibernate.validator.constraints.Range;

/**
 *
 */
@Message
public class PageForm {
    @Range(min = 0, max = 100)
    private int pageSize;
    private int page;

    public int getPageSize() {
        return pageSize;
    }

    public void setPageSize(int pageSize) {
        this.pageSize = pageSize;
    }

    public int getPage() {
        return page;
    }

    public void setPage(int page) {
        this.page = page;
    }

    @Override
    public String toString() {
        return "PageForm{" +
                "pageSize=" + pageSize +
                ", page=" + page +
                '}';
    }
}

File: server/manager/agent-js-sdk/form/PageForm.js
'use strict';



/**

*/
class PageForm {

    pageSize:Number;

    page:Number;
    constructor() {

    }

    formObject({pageSize,page}):PageForm{
        this.pageSize = pageSize;
        this.page = page;
        return this;
    }

    static of({pageSize,page}):PageForm{
        return new PageForm().formObject({pageSize,page});
    }

    static form(pageSize:Number,page:Number):PageForm{
        return new PageForm().formObject({pageSize,page});
    }
}

export default PageForm;